### ISSUE TITLE
hostname resource fails on SLES 11.4

### ISSUE THREAD

**Issue by robinedser** [chef-cookbooks/chef_hostname#41]
### Cookbook version
0.6.1

### Chef-client version
12.21.3

### Platform Details
SuSE 11.4

### Scenario:
Trying to set hostname on SLES 11 SP4.

### Steps to Reproduce:
Use hostname resource from chef_hostname to set hostname on SLES 11.4 system.

### Expected Result:
Hostname is set to provided value as with other platforms.

### Actual Result:
Chef client fails with error:

```
    ================================================================================
    Error executing action `set` on resource 'hostname[sles11test.standardbank.co.za]'
    ================================================================================
    
    Errno::EISDIR
    -------------
    Is a directory @ io_fread - /etc/sysconfig/network
    
    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/chef_hostname/resources/hostname.rb:14:in `read'
    /var/chef/cache/cookbooks/chef_hostname/resources/hostname.rb:14:in `append_replacing_matching_lines'
    /var/chef/cache/cookbooks/chef_hostname/resources/hostname.rb:100:in `block in class_from_file'
    /var/chef/cache/cookbooks/chef_hostname/resources/hostname.rb:202:in `block in after_created'
    /var/chef/cache/cookbooks/chef_hostname/resources/hostname.rb:201:in `each'
    /var/chef/cache/cookbooks/chef_hostname/resources/hostname.rb:201:in `after_created'

```

### Workaround / More information
This seems to be failing because on SLES 11.4 /etc/sysconfig/network is a directory and not a file (as on other distributions). The method **File.exist?** only checks whether the file is there, not whether the file is a directory or not (in Linux / Unix everything is a "file":

```
      when ::File.exist?("/etc/sysconfig/network")
        # older non-systemd RHEL/Fedora derived
        append_replacing_matching_lines("/etc/sysconfig/network", /^HOSTNAME\s*=/, "HOSTNAME=#{new_resource.hostname}")

```
I managed to get it to work by changing the method to use **File.file?** instead but have no idea if this is the correct way to fix this (new to chef & ruby):

```
      when ::File.file?("/etc/sysconfig/network")
        # older non-systemd RHEL/Fedora derived
        append_replacing_matching_lines("/etc/sysconfig/network", /^HOSTNAME\s*=/, "HOSTNAME=#{new_resource.hostname}")
```

Thank you for the resource it works brilliantly on the other platforms we use (RHEL7, CentOS7, SLES12) solving what you'd think would be a simple problem by ensuring our nodes' hostnames are consistently and correctly set.

**Comment by iennae** [comment]
Thanks for letting us know. The fix has been merged and the next release of the cookbook will resolve this.

**Comment by robinedser** [comment]
Awesome! :) Thanks Jennifer!

**Comment by tas50** [comment]
This is released in 1.0
